package.json

```diff
@@ -16,6 +16,7 @@
     "build:ts": "tsc",
     "build:ts:defs": "tsc --declaration --emitDeclarationOnly",
     "build:watch": "tsc --watch",
+    "repl": "ts-node -D 7017 -D 2393",
```

Comment:
This actually works pretty nice with `neoterm` and the right `.vimrc`.  Note that there's a couple of issues:

- [ts-node missing a recovery code](https://github.com/TypeStrong/ts-node/pull/1014)
- `ts-node` will still fail in annoying ways for some multiline expressions.  You basically want to turn off format on save for `.repl.ts`

Here's my current project hackety hack vimrc as it relates to this
```vim
" exploring repl workflows
" TODO: don't autoformat the repl file
"   requires adding 1007 to ts-node recovery codes
"     https://github.com/TypeStrong/ts-node/pull/1014
"   but this is still not enough due to `gh.pulls.list({` giving the wrong
"   error
" 
" Right now depends on manually doing
"   :2Tnew
"   " check the id of the new terminal
"   :Tls
"   :TREPLSetTerm id_of_repl_term
" 
function s:EditREPL()
  " setup REPL
  "   if REPL setup
  "     ensure visible
  "   else
  "   - C-W-t (go top left)
  "   - C-W n (split)
  "   - get bufnr
  "   - get term id from bufnr
  "   - TREPLSetTerm id
  "   - open repl file
  "
  exe "edit .repl.ts"
endfunction

nmap <leader>re :call <SID>EditREPL()<CR>
vmap <leader>re :TREPLSendSelection<CR>

augroup REPL
  au!
  au BufWritePre .repl.ts exe 'TREPLSendFile'
augroup END

```